### PR TITLE
Enable vision models for (Together, Fireworks, Meta-Reference, Ollama)

### PR DESCRIPTION
**Pull request by ashwinb**
# What does this PR do?

Adds support for vision models in the llama stack! We have claimed we support it but in actuality it didn't. Terrible.

For ollama, I tested with the 0.4.0 pre-release version. The tests passed. Once we have full fledged evals, we can better quantify "how well" things work with various providers.

## Test Plan

Added an integration test.

For meta-reference:
```bash
torchrun $CONDA_PREFIX/bin/pytest -v -s -m meta_reference test_vision_inference.py
```

For other fixtures (fireworks, ollama, together):
```bash
pytest -v -s -m fireworks test_vision_inference.py --env FIREWORKS_API_KEY=<..>
```

### CODE REVIEW

File: llama_stack/providers/adapters/inference/fireworks/fireworks.py [hunk 1]
```diff
@@ -129,32 +131,55 @@ async def _nonstream_chat_completion(
...
         return process_chat_completion_response(r, self.formatter)
 
     async def _stream_chat_completion(
         self, request: ChatCompletionRequest, client: Fireworks
     ) -> AsyncGenerator:
         params = self._get_params(request)
 
-        stream = client.completion.acreate(**params)
+        if "messages" in params:
+            print(f"Using chat completion endpoint: {params}")
```
Comment (raghotham) [inline llama_stack/providers/adapters/inference/fireworks/fireworks.py:146]: do we need this print? we need a broader debug/info/warn/error logging scheme

**Review by raghotham — APPROVED** [review]

File: llama_stack/providers/adapters/inference/ollama/ollama.py [hunk 2]
```diff
@@ -183,26 +172,66 @@ async def chat_completion(
...
+                ]
+            else:
+                input_dict["raw"] = True
+                input_dict["prompt"] = chat_completion_request_to_prompt(
+                    request, self.formatter
+                )
+        else:
+            assert (
+                not media_present
+            ), "Ollama does not support media for Completion requests"
```
Comment (ashwinb) [inline llama_stack/providers/adapters/inference/ollama/ollama.py:204]: this is technically not true. It actually does. I should enable this.

File: llama_stack/providers/impls/meta_reference/inference/inference.py [hunk 3]
```diff
@@ -87,6 +92,7 @@ async def completion(
             logprobs=logprobs,
         )
         self.check_model(request)
+        request = await request_with_localized_media(request)
```
Comment (yanxi0830) [inline llama_stack/providers/impls/meta_reference/inference/inference.py:95]: This seems to break if client uploads image via URL. 

```
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 421, in request_with_localized_media
    m.content = await _convert_content(m.content)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 415, in _convert_content
    return [await _convert_single_content(c) for c in content]
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 415, in <listcomp>
    return [await _convert_single_content(c) for c in content]
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/inline/inference/meta_reference/inference.py", line 408, in _convert_single_content
    url = await convert_image_media_to_url(content, download=True)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/llama_stack/providers/utils/inference/prompt_adapter.py", line 77, in convert_image_media_to_url
    r = await client.get(media.image.uri)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 1814, in get
    return await self.request(
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 1572, in request
    request = self.build_request(
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 346, in build_request
    url = self._merge_url(url)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_client.py", line 376, in _merge_url
    merge_url = URL(url)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_urls.py", line 117, in __init__
    self._uri_reference = urlparse(url, **kwargs)
  File "/home/xiyan/.conda/envs/llamastack-meta-reference-gpu/lib/python3.10/site-packages/httpx/_urlparse.py", line 158, in urlparse
    raise InvalidURL("URL too long")
httpx.InvalidURL: URL too long
```

Comment (ashwinb) [reply llama_stack/providers/impls/meta_reference/inference/inference.py:95]: @yanxi0830 I think the image becomes too large and is shipped as a "data" URL. This seems important to resolve. Filing an issue.

**Pull request merged**